### DISCUSSION TITLE
License for docker container

### DISCUSSION THREAD

**Discussion by kevinzezel** [dashingsoft/pyarmor#1321] [General]
I recently acquired the pro license, but I'm thinking I bought the wrong one. My software consists of several docker containers. We work with facial recognition and for example, for each security camera I have 1 docker container responsible for decoding the video. I have other containers that process faces and match the database. Therefore, for a client with 1000 cameras I will have more than 1000 containers running on different servers within the cluster at the same time. I need to obfuscate the source code in every docker container.

Is the pro license the most suitable? Or would it be the group? 

I haven't used the pro license yet, can I switch to the group license if necessary?

Regards,
Kevin

**Comment by jondy** [comment]
If you only run obfuscated scripts in 1000 docker containers, it's no problem for Pro license.

If you need obfuscate scripts in more than 100 docker containers every day, it need Group license, note that supporting unlimited docker feature is developing, it's not released.

According to Pyarmor EULA, if this license is not activated and purchased in 30 days, you can ask for refund, refer to
https://pyarmor.readthedocs.io/en/v8.2.6/licenses.html#refund-policy

**Reply by kevinzezel** [reply]
Each docker container has its own source code, so that all of them form the final software. Can I obfuscate everyone with the same license? Or will I need to have a license for each different container even if it belongs to the same software?

Ex:
Container 1: Decoder -> scale up to 100 containers
Container 2: Face Detection -> scale up to 5 containers
Container 3: Embeddings Extractions -> scale up to 5 containers
Container 4: Match faces -> scale up to 5 containers
Etc...

All of them form the final software.

Regards,
Kevin

**Reply by jondy** [reply]
one license is OK. But I don't think you need Group License, for example, in Container 1, it's decoder. Just obfuscate Decoder scripts in one container and copy the obfuscated scripts to others.

But if you have 100 different Containers, it may need group license. for example, 
...
Container 101: feature 101
Container 102: feature 102
...
